### ISSUE TITLE
[TT-6838] global_headers injection doesn't work after looping using target_url

### ISSUE THREAD

**Issue by armujahid** [TykTechnologies/tyk#4357]
**Branch/Environment/Version**
- Branch/Version: v4.2.2 (tykio/tyk-gateway:v4.2.2)
- Environment: local docker (and k8s)

**Describe the bug**
global_headers inject doesn't work after looping using target_url

Reference: https://community.tyk.io/t/global-request-headers-injection-doesnt-work-after-looping

**Reproduction steps**

1.  Use these 2 API definitions
API1
```json
{
  "name": "Tyk Keyless Looping API",
  "api_id": "keyless-looping-test",
  "org_id": "default",
  "definition": {
    "location": "header",
    "key": "version"
  },
  "use_keyless": true,
  "version_data": {
    "not_versioned": true,
    "versions": {
      "Default": {
        "name": "Default",
        "global_headers": {
          "X-looping1": "someValue"
        }
      }
    }
  },
  "driver": "otto",
  "disable_rate_limit": true,
  "proxy": {
    "listen_path": "/keyless-looping-test/",
    "target_url": "tyk://keyless-test",
    "strip_listen_path": true
  }
}

```
API2
```json
{
  "name": "Tyk Test Keyless API",
  "api_id": "keyless-test",
  "org_id": "default",
  "definition": {
    "location": "header",
    "key": "version"
  },
  "use_keyless": true,
  "version_data": {
    "not_versioned": true,
    "versions": {
      "Default": {
        "name": "Default",
        "global_headers": {
          "X-looping2": "ThisWillNotBeInjected"
        }
      }
    }
  },
  "driver": "otto",
  "disable_rate_limit": true,
  "proxy": {
    "listen_path": "/keyless-test/",
    "target_url": "http://httpbin.org",
    "strip_listen_path": true
  }
}

```

2. API1 is looping to API2.
3. API1 can be called using curl -s 'http://localhost:8080/keyless-looping-test/get' (tyk running on port 8080)

**Actual behavior**
In output we can see that header from API1 is injected but API2 failed to inject any header
Output:
```json
{
  "args": {}, 
  "headers": {
    "Accept": "*/*", 
    "Accept-Encoding": "gzip", 
    "Host": "httpbin.org", 
    "User-Agent": "curl/7.81.0", 
    "X-Amzn-Trace-Id": "Root=1-634fa839-5451453109031d7c04505e55", 
    "X-Looping1": "someValue"
  }, 
  "origin": "172.28.0.1, externalip", 
  "url": "http://httpbin.org/get"
}
```

**Expected behavior**
Global headers from API2 should be injected after looping from API1 to API2

**Screenshots/Video**
N/A

**Logs (debug mode or log file):**
N/A

**Configuration (tyk config file):**
Default config can be used

**Additional context**
1. Note that header injection from API2 works fine If I remove looping and replace `tyk://keyless-test` in API1 to `http://localhost:8080/keyless-test`
2. If `url_rewrites` is used instead of `target_url` as indicated by Olu in https://community.tyk.io/t/global-request-headers-injection-doesnt-work-after-looping/5841/2?u=armujahid then header injections starts to work. (Although it requires more boilerplate and separate entry for each HTTP verb?)

**Comment by armujahid** [comment]
This seems to be resolved. Tested in v4.3.2. I am now getting expected output with both global headers.
```
{
  "args": {}, 
  "headers": {
    "Accept": "*/*", 
    "Accept-Encoding": "gzip", 
    "Host": "httpbin.org", 
    "User-Agent": "curl/7.81.0", 
    "X-Amzn-Trace-Id": "Root=1-63e5e84f-1dc2eda57f767c552e90dd4c", 
    "X-Looping1": "someValue", 
    "X-Looping2": "ThisWillNotBeInjected"
  }, 
  "origin": "192.168.128.1, externalip", 
  "url": "http://httpbin.org/get"
}

```

**Comment by oluwaseyeayinla** [comment]
Closing. Fixed in v4.2.4
